src/core/rawlog.c

```diff
@@ -61,12 +62,8 @@ void rawlog_destroy(RAWLOG_REC *rawlog)
 /* NOTE! str must be dynamically allocated and must not be freed after! */
 static void rawlog_add(RAWLOG_REC *rawlog, char *str)
 {
-	if (rawlog->nlines < rawlog_lines || rawlog_lines <= 2)
-		rawlog->nlines++;
-	else {
-		void *tmp = rawlog->lines->data;
-		rawlog->lines = g_slist_remove(rawlog->lines,
-					       rawlog->lines->data);
+	if (rawlog->lines->length >= rawlog_lines && rawlog_lines > 0) {
```

Comment:
Maybe make this a while?  So rawlog_lines can be reduced without needing to reconnect to have it apply to existing rawlogs.